src/main/kotlin/com/openlattice/organizations/ExternalDatabaseManagementService.kt

```diff
@@ -376,10 +505,19 @@ class ExternalDatabaseManagementService(
             if (privileges.contains(PostgresPrivileges.INSERT) || privileges.contains(PostgresPrivileges.UPDATE))
                 permissions.add(Permission.WRITE)
             aces.executeOnKey(aceKey, PermissionMerger(permissions, objectType, OffsetDateTime.MAX))
+            return@map Acl(aclKeyUUIDs, setOf(Ace(principal, permissions, Optional.empty())))
         }
     }
 
     /*PRIVATE FUNCTIONS*/
+    private fun getNameFromFqnString(fqnString: String): String {
+        return fqnString.split(".")[1]
```

Comment:
You could also create a new `FullQualifiedName` here and then return its name. That way, it checks if it's not a malformed fqn.
`return FullQualifiedName(fqnString).name`

Reply:
Oh good idea